python/src/aac/context/definition_parser.py

```diff
@@ -9,735 +9,828 @@
+            lexemes (List[Lexeme]): A list of definition Lexemes.
+            defining_definition (Definition): Definition containing the field being checked.
 
-            # now add the fields to the class
-            for field in schema_definition.structure["schema"]["fields"]:
-                field_name = field["name"]
-                field_type = field["type"]
-                is_list = False
-
-                clean_field_type = field_type
-                if field_type.endswith("[]"):
-                    is_list = True
-                    clean_field_type = field_type[:-2]
-                if "(" in clean_field_type:
-                    clean_field_type = clean_field_type[: clean_field_type.find("(")]
-
-                # let's make sure the type of the field is known, or create it if it's not
-                potential_definitions = find_definitions_by_name(clean_field_type)
-                if len(potential_definitions) != 1:
-                    if len(potential_definitions) == 0:
+        Returns:
+            The field value.
+        """
+        python_type = defining_definition.structure["primitive"]["python_type"]
+        if is_list:
+            if not field_value:
+                if is_required:
+                    raise LanguageError(message=f"Missing required field {field_name}.", location=None)
+                field_value = []
+            else:
+                for item in field_value:
+                    if not isinstance(item, eval(python_type)):
                         raise LanguageError(
-                            f"Could not find AaC definition for type {clean_field_type} while loading {schema_definition.name}",
-                            get_location_str(field_type, schema_definition.lexemes),
+                            message=f"Invalid value for field '{field_name}'.  Expected type '{python_type}', but found '{type(item)}'",
+                            location=self.get_location_str(field_value, lexemes),
                         )
-                    else:
+        else:
+            if not field_value:
+                if is_required:
+                    raise LanguageError(f"Missing required field {field_name}.")
+            else:
+                if "Any" != python_type:
+                    if not isinstance(field_value, eval(python_type)):
                         raise LanguageError(
-                            f"Discovered multiple AaC definitions for type {clean_field_type} while loading {schema_definition.name}.  You may need to add a package name to differentiate.",
-                            get_location_str(field_type, schema_definition.lexemes),
+                            f"Invalid value for field '{field_name}'.  Expected type '{python_type}', but found '{type(field_value)}'",
+                            self.get_location_str(field_value, lexemes),
                         )
+        return field_value
```

Comment:
There should be a way to reduce the nesting here because the initial check to raise the language error is the same for the top level if check of trying to determine if it is a required field. 

I think part of the drive for how this nesting ended up this way is that list check, it should be able to be a separate one for allowing `field_value` to be set to the empty list. 

It will take some harder eval of the logic happening in each of the checks, but there are similar returns and raise of the Language error going on so it should be able to be simplified and condensed.